arangod/Cache/Table.cpp

```diff
@@ -211,25 +210,20 @@ Table::~Table() {
   }
 }
 
-std::uint64_t Table::allocationSize(std::uint32_t logSize) {
-  return sizeof(Table) +
-         (BUCKET_SIZE * (static_cast<std::uint64_t>(1) << logSize)) +
-         Table::padding;
-}
-
-std::uint64_t Table::memoryUsage() const {
-  return Table::allocationSize(_logSize);
+std::uint64_t Table::memoryUsage() const noexcept {
+  return allocationSize(_logSize);
 }
 
-std::uint64_t Table::size() const { return _size; }
+std::uint64_t Table::size() const noexcept { return _size; }
 
-std::uint32_t Table::logSize() const { return _logSize; }
+std::uint32_t Table::logSize() const noexcept { return _logSize; }
 
 Table::BucketLocker Table::fetchAndLockBucket(std::uint32_t hash,
                                               std::uint64_t maxTries) {
+  BucketLocker bucketGuard;
+
   SpinLocker guard(SpinLocker::Mode::Read, _lock,
                    static_cast<std::size_t>(maxTries));
```

Comment:
No, I don't think this code change had an effect on the lock order. Constructing a `BucketLocker` with arguments does not acquire any locks. It simplify creates a BucketLocker that hasn't locked anything. This can be seen by not passing any `this` or other `Table` pointer to the BucketLocker object. An actual BucketLocker object that holds a lock is later move-assigned to the `bucketGuard` object under some conditions. This position hasn't changed.
Also, the `bucketGuard` instance is later returned by the method, so it can still outlive the `SpinLocker guard` object as it could before. IMO this change is safe.